tket/src/Converters/ChoiMixTableauConverters.cpp

```diff
@@ -30,637 +30,730 @@ ChoiMixTableau circuit_to_cm_tableau(const Circuit& circ) {
-  std::map<Qubit, std::optional<unsigned>> in_z_row;
-  boost::bimap<Qubit, Qubit> matched_qubits;
-
-  // Call diagonalisation methods to diagonalise post-selected subspace
-  std::list<SpSymPauliTensor> to_diag;
-  for (unsigned r = tab.get_n_rows(); r > 0;) {
-    --r;
-    ChoiMixTableau::row_tensor_t rten = tab.get_row(r);
-    if (rten.second.size() != 0) {
-      // Reached the rows with non-empty output segment
-      break;
-    }
-    // Else, we add the row to the subspace
-    to_diag.push_back(rten.first);
+  for (const Qubit& init_q : unitary_init_names) {
+    if (tab.col_index_.left.find(ChoiMixTableau::col_key_t{
+            init_q, ChoiMixTableau::TableauSegment::Input}) !=
+        tab.col_index_.left.end())
+      throw std::logic_error(
+          "Free qubit name for initialisation conflicts with existing live "
+          "input of ChoiMixTableau");
   }
-  unsigned post_selected_size = to_diag.size();
-  std::set<Qubit> diag_ins{input_qubits.begin(), input_qubits.end()};
-  Circuit in_circ = mutual_diagonalise(to_diag, diag_ins, CXConfigType::Tree);
-  // Extract the dagger of each gate in order from tab
-  for (const Command& com : in_circ) {
-    auto args = com.get_args();
-    qubit_vector_t qbs = {args.begin(), args.end()};
-    tab.apply_gate(
-        com.get_op_ptr()->dagger()->get_type(), qbs,
-        ChoiMixTableau::TableauSegment::Input);
+  for (const Qubit& post_q : unitary_post_names) {
+    if (tab.col_index_.left.find(ChoiMixTableau::col_key_t{
+            post_q, ChoiMixTableau::TableauSegment::Output}) !=
+        tab.col_index_.left.end())
+      throw std::logic_error(
+          "Free qubit name for post-selection conflicts with existing live "
+          "output of ChoiMixTableau");
   }
+}
 
-  // Diagonalised rows should still be at the bottom of the tableau - reduce
-  // them to a minimal set of qubits for post-selection by first reducing to
-  // upper echelon form
-  std::vector<std::pair<unsigned, unsigned>> row_ops =
-      gaussian_elimination_row_ops(
-          tab.tab_.zmat_.bottomRows(post_selected_size));
-  for (const std::pair<unsigned, unsigned>& op : row_ops) {
-    tab.tab_.row_mult(op.first, op.second);
-  }
-  // Obtain CX instructions as column operations
-  std::vector<std::pair<unsigned, unsigned>> col_ops =
-      gaussian_elimination_col_ops(
-          tab.tab_.zmat_.bottomRows(post_selected_size));
-  // These gates will also swap qubits to isolate the post-selections on the
-  // first few qubits - this is fine as can be cleaned up later with peephole
-  // optimisations
-  for (const std::pair<unsigned, unsigned>& op : col_ops) {
-    tab.tab_.apply_CX(op.first, op.second);
-    ChoiMixTableau::col_key_t ctrl = tab.col_index_.right.at(op.first);
-    ChoiMixTableau::col_key_t trgt = tab.col_index_.right.at(op.second);
-    in_circ.add_op<Qubit>(OpType::CX, {ctrl.first, trgt.first});
+// DEBUG METHOD: Ignore this for coverage checks
```

Comment:
Are we ignoring it?

Reply:
This comment was more intended for any developer who is scanning through coverage reports to check for ways to improve their coverage. I don't know of any simple method for ignoring methods from gcovr - if you have any suggestions please let me know! I'd rather not have to lift this entire class structure up to a header and move this to test files, as the class is just a helper utility to perform a single method

Reply:
Ah I see. You should be able to use exclusion markers: https://gcovr.com/en/stable/guide/exclusion-markers.html .

Reply:
Thanks for the suggestion, I have added the markers and checked it ignores it locally. The CI errors seem to be from a quimb update which looks independent from this PR and will need some looking into

Reply:
> Thanks for the suggestion, I have added the markers and checked it ignores it locally. The CI errors seem to be from a quimb update which looks independent from this PR and will need some looking into

Ah yes, we had this issue on the pytket-docs repo as well. Should get this fixed ASAP.